From e17a3590ef7a4ad10e097212748d0f514115a107 Mon Sep 17 00:00:00 2001
Message-Id: <e17a3590ef7a4ad10e097212748d0f514115a107.1350045285.git.minovotn@redhat.com>
In-Reply-To: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
References: <8d140e6fe679a73fd193609be44d508f17f7892b.1350045285.git.minovotn@redhat.com>
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 11 Oct 2012 18:24:24 +0200
Subject: [PATCH 12/12] blockdev: drive_reopen(): pass error message to
 OpenFileFailed error

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1349979864-509-7-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 42996
O-Subject: [RHEL6.4 qemu-kvm PATCH 6/6] blockdev: drive_reopen(): pass error message to OpenFileFailed error
Bugzilla: 806775
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Pavel Hrdina <phrdina@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 blockdev.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 blockdev.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/blockdev.c b/blockdev.c
index 4fc76f5..deecfa1 100644
--- a/blockdev.c
+++ b/blockdev.c
@@ -769,11 +769,14 @@ void qmp___com_redhat_drive_reopen(const char *device, const char *new_image_fil
      * are in serious trouble.
      */
     if (ret != 0) {
-        ret = bdrv_open(bs, old_filename, flags, old_drv);
-        if (ret != 0) {
-            error_set(errp, QERR_OPEN_FILE_FAILED, old_filename, "");
+        int ret2;
+        ret2 = bdrv_open(bs, old_filename, flags, old_drv);
+        if (ret2 != 0) {
+            error_set(errp, QERR_OPEN_FILE_FAILED, old_filename,
+                      strerror(-ret2));
         } else {
-            error_set(errp, QERR_OPEN_FILE_FAILED, new_image_file, "");
+            error_set(errp, QERR_OPEN_FILE_FAILED, new_image_file,
+                      strerror(-ret));
         }
     }
 }
-- 
1.7.11.7

